akka-persistence/src/main/scala/akka/persistence/journal/AsyncWriteJournal.scala

```diff
@@ -47,7 +47,8 @@ trait AsyncWriteJournal extends Actor with WriteJournalBase with AsyncRecovery {
   private def isReplayFilterEnabled: Boolean = replayFilterMode != ReplayFilter.Disabled
   private val replayFilterWindowSize: Int = config.getInt("replay-filter.window-size")
   private val replayFilterMaxOldWriters: Int = config.getInt("replay-filter.max-old-writers")
-  private val replayDebugEnabled: Boolean = config.getBoolean("replay-filter.debug")
+  // cannot be a val due to binary compatibility
+  private def replayDebugEnabled: Boolean = config.getBoolean("replay-filter.debug")
```

Comment:
Yes, that would be possible, I think. I didn’t consider accessing the config to be expensive, but now some memories are coming back that it actually was beneficial to cache these lookups. Will change.